src/io/syncfile.c

```diff
@@ -492,6 +492,18 @@ MVMObject * MVM_file_open_fh(MVMThreadContext *tc, MVMString *filename, MVMStrin
         MVM_exception_throw_adhoc_free(tc, waste, "Failed to open file %s: %s", fname, uv_strerror(req.result));
     }
 
+    /* Check that we didn't open a directory by accident. If stat fails, just move on. */
```

Comment:
I also managed to introduce a bug by trying to close `data->fd` instead of `fd` :(

I'll add the comment and fix the other stuff after lunch.